modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/loading/Retry.scala

```diff
@@ -58,19 +60,22 @@ object Retry {
         case Strategy.Exponential => RetryPolicies.exponentialBackoff[F](retries.backoff)
       }
 
-      val withAttempts = retries.attempts match {
+      retries.attempts match {
         case Some(attempts) =>
           policy.join(RetryPolicies.limitRetries(attempts))
         case None =>
           policy
       }
+    }
 
-      retries.cumulativeBound match {
-        case Some(bound) =>
-          RetryPolicies.limitRetriesByCumulativeDelay(bound, withAttempts)
-        case None =>
-          withAttempts
-      }
+  def isWithinCumulativeBound[F[_]: Monad: Clock](config: Retries, started: FiniteDuration): F[Boolean] =
```

Comment:
What do you think about using this new `isWithinCumulativeBound` only in one place, here in a custom `RetryPolicy`? If it's doable, then we wouldn't have to change much of a code on a caller side. I'm thinking about sth like this: 

```scala

  def getRetryPolicy[F[_]: Applicative: Clock](retries: Retries): F[RetryPolicy[F]] =
    Clock[F].realTime { startedAt =>

      ... other policies...
      
        val withCumulativeBound = RetryPolicy { _ =>
          isWithinCumulativeBound[F](retries, startedAt).ifF(
            ifTrue = PolicyDecision.DelayAndRetry(Duration.Zero),
            ifFalse = PolicyDecision.GiveUp
          )
        }
        
        otherPolicies.join(withCumulativeBound)
      }
```

Then `getRetryPolicy` would have slightly different signature.

Reply:
I like that.  I'll try it out.